core/trino-spi/src/main/java/io/trino/spi/eventlistener/QueryPlanOptimizerStatistics.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.eventlistener;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is JSON serializable for convenience and serialization compatibility is not guaranteed across versions.
+ */
+public record QueryPlanOptimizerStatistics(String rule, long invocations, long applications, long totalTime, long failures)
+{
+    @JsonCreator
+    public QueryPlanOptimizerStatistics(
+            @JsonProperty("rule") String rule,
+            @JsonProperty("invocations") long invocations,
+            @JsonProperty("applications") long applications,
+            @JsonProperty("totalTime") long totalTime,
+            @JsonProperty("failures") long failures)
```

Comment:
You should be able to annotate the record declaration directly instead of needing an explicit constructor.